.github/workflows/rust.yml

```diff
@@ -11,11 +11,21 @@ env:
 
 jobs:
   build:
-
-    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # macOS 12, Xcode: 14.2
+        # macOS 13, Xcode: 15.2
+        # macOS 14, Xcode: 15.3
```

Comment:
macOS 14 would would be subject to change over time. It'd be probably more helpful to comment this matrix including the macOS SDK versions used.